### PR TITLE
Feature/form

### PR DESCRIPTION
**Pull request by PaigeVannelli**
## Is this a fix or a feature?
Feature 

## What is the change?
Set up a reusable Form component for user inputs allowing a user to add name, email, username, make, model, year, origin, and destination. Added error handling so a user cannot advance or enter the form until all inputs are completed. Connect the next button so that it advances to the next set of inputs. 

## What does it fix?
n/a

## Where should the reviewer start?
Review App.tsx and Form.tsx to see form changes 

## How should this be tested?
Open app with npm start and advance through form inputs. Check error handling by trying to click the next button without filling out all forms.

### CODE REVIEW

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          header="days and time"
+          page={page}
+          setPage={setPage}
+          inputs={[{property: destination, method: setDestination, placeholder: 'destination'}]} 
+        />
+      }
+    </div>
+  );
+}
+
```
Comment (itsnameissteven) [inline src/components/App/App.tsx:60]: I like this setup... we talked a little bit about it already but we may eventually want to have all this conditional rendering happen in another component that contains all forms. 

We would have to pass down all the state and methods, which is a lot. OR we can go wild and mess with the context api.

File: src/components/Form/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  inputs: Array<Inputs>
+}
+
+interface Inputs {
+  property: string
+  method: React.Dispatch<React.SetStateAction<string>>
+  placeholder: string
+}
+
+const Form: React.FC<FormProps> = ({ header, page, setPage, inputs }): JSX.Element => {
```
Comment (itsnameissteven) [inline src/components/Form/Form.tsx:16]: I like the interfaces they are clean and clear. One things we might want to consider is restructuring react types at the top of the file. eg React.Dispatch, React.FC ...etc

Im not 100% but the import at the top would look something like this:

`import React, { FC, Dispatch, SetStateAction, FormEvent } from 'react';`

it wouldn't change much, but we wouldn't have 'React.' a bunch of times

**Review by itsnameissteven — APPROVED** [review]
Great job getting the ball rolling on this. TypeScript is looking clean and functional!

File: src/components/Form/Form.tsx (hunk 2 above)
Comment (alexthompson207) [reply src/components/Form/Form.tsx:16]: That's a good call, I haven't seen that before but I like it!

File: src/components/App/App.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const [name, setName] = useState('')
+  const [username, setUsername] = useState('')
+  const [email, setEmail] = useState('')
+  const [page, setPage] = useState(0)
+  const [make, setMake] = useState('')
+  const [model, setModel] = useState('')
+  const [year, setYear] = useState('')
+  const [origin, setOrigin] = useState('')
+  const [destination, setDestination] = useState('')
```
Comment (alexthompson207) [inline src/components/App/App.tsx:16]: I wonder if there is a way to condense this somehow? Maybe for a future refactor. Otherwise this looks great!

**Review by alexthompson207 — APPROVED** [review]
Way to start off the project! This is an awesome start, good work!!

**Pull request merged**